### ISSUE TITLE
ScalarGE fails to converge

### ISSUE THREAD

**Issue by Jilllllscout** [XMU-Kuangnan-Fang-Team/GENetLib#21]
Hi GENetLib team,
I've been exploring the capabilities of GENetLib, specifically the ScalarGE model, for my research on gene-environment interactions. I'm quite impressed with the functionality and the potential applications. However, I've encountered an issue that I believe is worth addressing.
When running the ScalarGE model with the default parameter settings provided in the example code, I've noticed that the model fails to converge. This is a significant hurdle as it prevents me from obtaining meaningful results from the analysis. I understand that convergence issues can be sensitive to initial parameter settings, but I was under the impression that the default settings would offer a reasonable starting point for most use cases.

Dataset: I'm working with a dataset consisting of n = 1500 samples, with p = 500 genetic variables and q = 5 environmental variables.
Parameter Settings: I've used the parameters as outlined in the example code:
num_hidden_layers = 2
nodes_hidden_layer = [1000, 100]
num_epochs = 100
learning_rate1 = 0.06
learning_rate2 = 0.035
lambda1 = None
lambda2 = 0.09
Lambda = 0.1
threshold = 0.01

The model's loss function fluctuates erratically, and I've encountered NaN values in the parameter updates, leading to training termination.

System Information:
GENetLib version: 1.1.7
Python version: 3.8.5

I've tried adjusting the learning rates and regularization parameters slightly, but the issue persists. I've also ensured that my dataset is preprocessed correctly, with no missing values or outliers that could affect the model's performance.
I would appreciate any guidance on how to resolve this issue or if there are any recommended practices for parameter tuning in ScalarGE. If this is a known issue, I'd be interested in any updates or patches that address it.
Thank you for your attention to this matter. I'm looking forward to your response and hope to continue using GENetLib for my research.
Best regards

**Comment by Barry57** [comment]
Thank you for your interest in GENetLib. We have indeed encountered such issues in practice. The possible explanation is the instability of neural networks. We recommend using grid search to ensure that your parameters are optimized.

**Comment by ruiqwy** [comment]
Thank you for your attention. In the future, we will undertake some work on the stability of neural networks to address this issue.
